app/code/Magento/Quote/Model/QuoteIdToMaskedQuoteId.php

```diff
@@ -19,43 +21,70 @@ class QuoteIdToMaskedQuoteId implements QuoteIdToMaskedQuoteIdInterface
      * @var QuoteIdMaskFactory
      */
     private $quoteIdMaskFactory;
-    /**
-     * @var CartRepositoryInterface
-     */
-    private $cartRepository;
 
     /**
      * @var QuoteIdMaskResource
      */
     private $quoteIdMaskResource;
+    /**
+     * @var CartInterfaceFactory
+     */
+    private $cartFactory;
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
 
     /**
      * @param QuoteIdMaskFactory $quoteIdMaskFactory
-     * @param CartRepositoryInterface $cartRepository
      * @param QuoteIdMaskResource $quoteIdMaskResource
+     * @param CartInterfaceFactory $cartInterfaceFactory
+     * @param StoreManagerInterface $storeManager
      */
     public function __construct(
         QuoteIdMaskFactory $quoteIdMaskFactory,
-        CartRepositoryInterface $cartRepository,
-        QuoteIdMaskResource $quoteIdMaskResource
+        QuoteIdMaskResource $quoteIdMaskResource,
+        CartInterfaceFactory $cartInterfaceFactory,
+        StoreManagerInterface $storeManager
     ) {
         $this->quoteIdMaskFactory = $quoteIdMaskFactory;
-        $this->cartRepository = $cartRepository;
         $this->quoteIdMaskResource = $quoteIdMaskResource;
+        $this->cartFactory = $cartInterfaceFactory;
+        $this->storeManager = $storeManager;
     }
 
     /**
      * @inheritDoc
      */
     public function execute(int $quoteId): string
     {
-        /* Check the quote exists to avoid database constraint issues */
-        $this->cartRepository->get($quoteId);
+        // Check the quote exists to avoid database constraint issues
+        $this->checkIfQuoteExists($quoteId);
 
         $quoteIdMask = $this->quoteIdMaskFactory->create();
         $this->quoteIdMaskResource->load($quoteIdMask, $quoteId, 'quote_id');
-        $maskedId = $quoteIdMask->getMaskedId() ?? '';
 
-        return $maskedId;
+        return $quoteIdMask->getMaskedId() ?? '';
+    }
+
+    /**
+     * This is tiny implementation similar to \Magento\Quote\Api\CartRepositoryInterface::get
```

Comment:
OMG. MHI test failed.
![image](https://user-images.githubusercontent.com/1873745/123093637-9948b900-d434-11eb-94d7-67e54a41ad6c.png)

This is a false positive, but it's way easier to fix it like this:

```suggestion
     * This is a tiny implementation similar to the get method in \Magento\Quote\Api\CartRepositoryInterface
```

Reply:
@sidolov @sivaschenko @gabrieldagama, could you report an issue to fix such false positives in MHI?